src/react/AppProvider.jsx

```diff
@@ -61,7 +68,28 @@ export default function AppProvider({ store, children }) {
     setLocale(getLocale());
   });
 
-  const appContextValue = useMemo(() => ({ authenticatedUser, config, locale }), [authenticatedUser, config, locale]);
+  const [appThemeState, appThemeDispatch] = useAppTheme({
+    themeUrls: {
+      [APP_THEME_CORE]: config.APP_THEME_CORE_URL,
+      variants: {
+        [APP_THEME_LIGHT]: config.APP_THEME_LIGHT_URL,
+      },
+    },
+  });
+
+  const appContextValue = useMemo(() => ({
+    authenticatedUser,
+    config,
+    locale,
+    appTheme: {
+      state: appThemeState,
+      dispatch: appThemeDispatch,
+    },
+  }), [authenticatedUser, config, locale, appThemeState, appThemeDispatch]);
+
+  if (!appThemeState?.isThemeLoaded) {
+    return null;
+  }
```

Comment:
[inform] Don't render the app until the core theme CSS and theme variant(s) CSS is loaded.